packages/ui/src/components/DatePicker/Calendar/Calendar.tsx

```diff
@@ -0,0 +1,136 @@
+import ChevronLeftIcon from '@mui/icons-material/ChevronLeft';
+import ChevronRightIcon from '@mui/icons-material/ChevronRight';
+import * as React from 'react';
+
+import { Box } from '../../Box/Box';
+import { IconButton } from '../../IconButton';
+import { Stack } from '../../Stack/Stack';
+import { Typography } from '../../Typography/Typography';
+import { DayBox } from './Calendar.styles';
+
+import type { DateTime } from 'luxon';
+
+interface CalendarProps {
+  direction: 'left' | 'right';
+  endDate: DateTime | null;
+  focusedField: 'end' | 'start';
+  month: DateTime;
+  onDateClick: (date: DateTime, field: 'end' | 'start') => void;
+  setMonth: (date: DateTime) => void;
+  startDate: DateTime | null;
+}
+
+export const Calendar = ({
+  direction,
+  endDate,
+  focusedField,
+  month,
+  onDateClick,
+  setMonth,
+  startDate,
+}: CalendarProps) => {
+  const startOfMonth = month.startOf('month');
+  const endOfMonth = month.endOf('month');
+  const startDay = startOfMonth.weekday % 7;
+  const totalDaysInMonth = endOfMonth.day;
+  const totalGridCells = 42; // Always 6 rows (6 × 7)
+  const days = [];
+
+  // Fill leading empty slots before the first day of the month
+  for (let i = 0; i < startDay; i++) {
+    days.push(<Box key={`empty-${i}`} sx={{ height: 40, width: 40 }} />);
+  }
+
+  // Fill actual days of the month
+  for (let day = 1; day <= totalDaysInMonth; day++) {
+    const currentDay = month.set({ day });
+    const isSelected =
+      startDate &&
+      endDate &&
+      startDate.isValid &&
+      endDate.isValid &&
+      currentDay >= startDate &&
+      currentDay <= endDate;
+    const isStartOrEnd =
+      (startDate && startDate.isValid && currentDay.equals(startDate)) ||
+      (endDate && endDate.isValid && currentDay.equals(endDate));
+
+    days.push(
+      <DayBox
+        isSelected={isSelected}
+        isStartOrEnd={isStartOrEnd}
+        key={day}
```

Comment:
Nit: include the month in this key which fixes a slight flickering issue when scrolling between months:

|Before|After|
|-|-|
|<video src="https://github.com/user-attachments/assets/d43b6e59-874f-4bba-a6f4-4a6f226ba083" />|<video src="https://github.com/user-attachments/assets/ccbe6566-5520-48cc-8b3d-60660e180824" />|


```suggestion
        key={`${month.month} ${day}`}
```